src/callbacks_step/glm_speed.jl

```diff
@@ -83,7 +83,7 @@ function update_cleaning_speed!(semi, glm_speed_callback, dt)
     c_h_deltat = calc_dt_for_cleaning_speed(cfl, mesh, equations, solver, cache)
 
     # c_h is proportional to its own time step divided by the complete MHD time step
-    equations.c_h = glm_scale * c_h_deltat / dt
+    @reset equations.c_h = glm_scale * c_h_deltat / dt
```

Comment:
The problem with this is that you create a new `equations` struct locally that does not propagate any information to the one stored in `semi`. I do not have a perfect solution here. The only way this could work is if the `semi` was `mutable` and we changed the `equations` therein.

Reply:
Fair what about using `@reset` again for `semi`.

Reply:
Can run without errors except the `Warning: Instability detected. Aborting` if you are going to try these examples
https://github.com/trixi-framework/Trixi.jl/blob/68df09d5a21bd8f7393df90dab915247f9498505/examples/tree_2d_dgsem/elixir_mhd_ec.jl https://github.com/trixi-framework/Trixi.jl/blob/68df09d5a21bd8f7393df90dab915247f9498505/examples/tree_3d_dgsem/elixir_mhd_ec.jl
Do you know why? @ranocha

Reply:
The `SemidiscretizationHyperbolic` is still a `struct`, so you just create a copy of it with new `equations` but what is stored in the `ode` does not change

Reply:
I see